packages/collector/test/announceCycle/unannounced_test.js

```diff
@@ -0,0 +1,187 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ */
+
+'use strict';
+
+const chai = require('chai');
+const proxyquire = require('proxyquire');
+const sinon = require('sinon');
+const sinonChai = require('sinon-chai');
+
+const expect = chai.expect;
+chai.use(sinonChai);
+
+const { secrets, tracing } = require('@instana/core');
+const { constants } = tracing;
+
+let unannouncedState;
+let agentResponse;
+
+const pidStoreStub = {};
+const agentOptsStub = {};
+const tracingStub = sinon.stub(tracing);
+const secretsStub = sinon.stub(secrets);
```

Comment:
This won't work, because if you run the tests this code will be executed for **all tests**.
That means you would stub the whole core package for all tests.
The stub call needs to go into `before` or `beforeEach` - depends what you want to achieve :)